src/python/pants/testutil/engine/util.py

```diff
@@ -19,23 +21,46 @@
 from pants.util.objects import SubclassesOf
 
 
-def run_rule(rule, *args):
-  """A test helper function that runs an @rule with a set of arguments and Get providers.
+@dataclass(frozen=True)
+class MockedYieldGet:
```

Comment:
What happens with `yield [Get(..)]`? How do we mock that?

Reply:
This code:

https://github.com/pantsbuild/pants/blob/c39b7a4cb487bd074c42c42df88ebd62e97b4803/src/python/pants/testutil/engine/util.py#L59-L79

Specifically:

https://github.com/pantsbuild/pants/blob/c39b7a4cb487bd074c42c42df88ebd62e97b4803/src/python/pants/testutil/engine/util.py#L76-L77

The only thing that changes is our inner `get()` function's implementation, but it has the same API.

Reply:
So I think this means that this is a `MockGet`, not a `MockedYieldGet`. We're mocking the Get (which although a verb in English, is a noun in this context). Not the `Yield`, which is a verb in both English and this context...